Duplicate function track<Entity>ById when 2 relations with same entity

##### **Overview of the issue**

Using ng2, I have 2 entities MatchResult and Player.
MatchResult has 2 relations with Player: winner and loser.

The `match-result-dialog.component.ts` file cannot compile due to duplicate function `trackPlayerById`

##### **Motivation for or Use Case**

Compilation error.

~~~
yarn run tsc
yarn run v0.18.1
$ tsc
src/main/webapp/app/entities/match-result/match-result-dialog.component.ts(77,5): error TS2393: Duplicate function implementation.
src/main/webapp/app/entities/match-result/match-result-dialog.component.ts(80,5): error TS2393: Duplicate function implementation.
~~~

##### **Suggest a Fix**

Maybe the functions should be named according to relation name: `trackWinnerById` and `trackLoserById` 

##### **JHipster Version(s)**

latest master at commit 478eeb602b2c2f47d84329441541cf0d7561d897

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

~~~
{
  "generator-jhipster": {
    "jhipsterVersion": "3.12.1",
    "baseName": "match",
    "packageName": "com.example",
    "packageFolder": "com/example",
    "serverPort": "8100",
    "authenticationType": "session",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "rememberMeKey": "61a1b4c97a6472f60d101e218f9fd98f672e31f9",
    "useSass": false,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en"
    ],
    "serviceDiscoveryType": false,
    "clientFramework": "angular2"
  }
}
~~~

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

Player.json
~~~
{
    "fluentMethods": true,
    "relationships": [],
    "fields": [
        {
            "fieldName": "name",
            "fieldType": "String",
            "fieldValidateRules": [
                "required"
            ]
        }
    ],
    "changelogDate": "20161230222153",
    "dto": "no",
    "service": "no",
    "entityTableName": "player",
    "pagination": "no"
}
~~~

MatchResult.json
~~~
{
    "fluentMethods": true,
    "relationships": [
        {
            "relationshipName": "winner",
            "otherEntityName": "player",
            "relationshipType": "many-to-one",
            "otherEntityField": "name"
        },
        {
            "relationshipName": "loser",
            "otherEntityName": "player",
            "relationshipType": "many-to-one",
            "otherEntityField": "name"
        }
    ],
    "fields": [
        {
            "fieldName": "score",
            "fieldType": "String"
        }
    ],
    "changelogDate": "20161230222831",
    "dto": "no",
    "service": "no",
    "entityTableName": "match_result",
    "pagination": "pager"
}
~~~

--- a/generators/entity/templates/client/angular/src/main/webapp/app/entities/_entity-management-dialog.component.ts
+++ b/generators/entity/templates/client/angular/src/main/webapp/app/entities/_entity-management-dialog.component.ts
@@ -101,12 +101,14 @@ export class <%= entityAngularJSName %>DialogComponent implements OnInit {
         this.alertService.error(error.message, null, null);
     }
     <%_
-    for (idx in relationships) {
-        var otherEntityNameCapitalized = relationships[idx].otherEntityNameCapitalized; _%>
-    track<%- otherEntityNameCapitalized %>ById(index, item: <%- relationships[idx].otherEntityNameCapitalized %>) {
+    var entitiesSeen = [];
+    for (idx in relationships) { _%>
+    <%  var otherEntityNameCapitalized = relationships[idx].otherEntityNameCapitalized;
+        if(entitiesSeen.indexOf(otherEntityNameCapitalized) == -1) { %>
+    track<%- otherEntityNameCapitalized -%>ById(index, item: <%- relationships[idx].otherEntityNameCapitalized -%>) {
         return item.id;
     }
-    <%_ } _%>
+    <%_ entitiesSeen.push(otherEntityNameCapitalized); } } _%>
 
     <%_ if (hasManyToMany) { _%>
     getSelected(selectedVals: Array<any>, option: any) {

